<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Changelog • SamplingStrata</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Changelog"><meta property="og:image" content="/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-156503236-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-156503236-1');
</script></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-news">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SamplingStrata</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.5-4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/SamplingStrata.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/drake_continuous.html">SamplingStrata workflow with drake</a>
    </li>
    <li>
      <a href="../articles/models.html">Use of models in SamplingStrata</a>
    </li>
    <li>
      <a href="../articles/spatial.html">Spatial sampling with SamplingStrata</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/barcaroli/SamplingStrata/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
      <h1 data-toc-skip>Changelog <small></small></h1>
      <small>Source: <a href="https://github.com/barcaroli/SamplingStrata/blob/HEAD/NEWS.md" class="external-link"><code>NEWS.md</code></a></small>
    </div>

    <div class="section level2">
<h2 class="page-header" data-toc-text="1.5-4" id="samplingstrata-15-4">SamplingStrata 1.5-4<small>2022-11-15</small><a class="anchor" aria-label="anchor" href="#samplingstrata-15-4"></a></h2>
<ul><li>Minor changes due to the introduction of R 4.3.0</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.5-3" id="samplingstrata-15-3">SamplingStrata 1.5-3<small>2022-08-28</small><a class="anchor" aria-label="anchor" href="#samplingstrata-15-3"></a></h2>
<div class="section level3">
<h3 id="major-changes-1-5-3">Major changes<a class="anchor" aria-label="anchor" href="#major-changes-1-5-3"></a></h3>
<ul><li>As in R 4.2.0 a matrix objects now also inherit from class “array”, this invalidates code incorrectly assuming that class(matrix_obj)) has length one. So, changes have been done consequently in SamplingStrata functions where tests on class of an object were performed, otherwise causing errors.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.5-2" id="samplingstrata-15-2">SamplingStrata 1.5-2<small>2022-01-07</small><a class="anchor" aria-label="anchor" href="#samplingstrata-15-2"></a></h2>
<div class="section level3">
<h3 id="major-changes-1-5-2">Major changes<a class="anchor" aria-label="anchor" href="#major-changes-1-5-2"></a></h3>
<ul><li><p>A new function <em>selectSampleSpatial</em> has been added: if geographical coordinates are available in the frame, in order to obtain spatially distributed selected points this function makes use of the lpm2_kdtree function from the SamplingBigData package (Tillé-Grafstrom).</p></li>
<li><p>A new function <em>bethelProc</em> has been added: this function allows to execute a complete procedure from the Bethel optimal allocation to the selection of a sample, without having to optimize the strata, that are supposed to be given and fixed.</p></li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.5-1" id="samplingstrata-15-1">SamplingStrata 1.5-1<small>2020-03-02</small><a class="anchor" aria-label="anchor" href="#samplingstrata-15-1"></a></h2>
<div class="section level3">
<h3 id="major-changes-1-5-1">Major changes<a class="anchor" aria-label="anchor" href="#major-changes-1-5-1"></a></h3>
<ul><li><p>As starting from R 4.0.0 ‘stringsAsFactors = FALSE’ becomes a default, in all calls to data.frame the parameter ‘stringAsFactors = TRUE’ has been indicated, in order to ensure the same results</p></li>
<li><p>Fixed some bugs related to handling take-all strata for ‘atomic’ and ‘spatial’ methods</p></li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.5" id="samplingstrata-15">SamplingStrata 1.5<small>2020-01-23</small><a class="anchor" aria-label="anchor" href="#samplingstrata-15"></a></h2>
<div class="section level3">
<h3 id="major-changes-1-5">Major changes<a class="anchor" aria-label="anchor" href="#major-changes-1-5"></a></h3>
<ul><li><p>A new <em>optimStrata</em> function is available: this function is a wrapper that allows to execute the three different optimization functions: (i) optimizeStrata (method = <em>atomic</em>); (ii) optimizeStrata (method = <em>continuous</em>); (iii) optimizeStrata (method = <em>spatial</em>).</p></li>
<li><p>A new <em>optimizeStrataSpatial</em> function is available: this function optimizes the frame stratification taking into account also spatial correlation of frame units in a territorial context. As for <em>optimizeStrata2</em>, this function can be used only on continuous stratification variables.</p></li>
<li><p>A new <em>kMeansSolutionSpatial</em> function has been added: this function is the same than <em>KmeansSolution</em>, but operates in the case of optimization with only continuous stratification variables and has to be used only in conjunction with <em>optimizeStrataSpatial</em>.</p></li>
<li><p>A new <em>KmeansSolution2</em> function has been added: this function is the same than <em>KmeansSolution</em>, but operates in the case of optimization with only continuous stratification variables and has to be used only in conjunction with <em>optimizeStrata2</em>.</p></li>
<li><p>A new <em>prepareSuggestion</em> function has been added: this function operates on the result of <em>kMeanSolution2</em> or <em>kMeanSolutionSpatial</em> order to prepare the suggestions for the optimization with only continuous stratification variables.</p></li>
<li><p>A new <em>computeGamma</em> function has been added: this function allows to calculate a heteroscedasticity index, to be passed to optimization step as a <em>model</em> parameter in order to correctly evaluate the variance in the strata.</p></li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.4-1" id="samplingstrata-14-1">SamplingStrata 1.4-1<small>2019-04-22</small><a class="anchor" aria-label="anchor" href="#samplingstrata-14-1"></a></h2>
<div class="section level3">
<h3 id="major-changes-1-4-1">Major changes<a class="anchor" aria-label="anchor" href="#major-changes-1-4-1"></a></h3>
<ul><li><p>A new <em>summaryStrata</em> function has been added, enabling to get structured information regarding the strata produced by the <em>optimizeStrata2</em> function (operating on only continuous stratification variables).</p></li>
<li><p>A new <em>assignStrataLabel</em> function has been added, enabling to assign the optimized strata label to new units to be added in the sampling frame.</p></li>
<li><p>Fixed a bug in the optimization step for continuous variables</p></li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.4" id="samplingstrata-14">SamplingStrata 1.4<small>2019-01-02</small><a class="anchor" aria-label="anchor" href="#samplingstrata-14"></a></h2>
<div class="section level3">
<h3 id="major-changes-1-4">Major changes<a class="anchor" aria-label="anchor" href="#major-changes-1-4"></a></h3>
<ul><li><p>A new function <em>optimizeStrata2</em> is available. This function performs the same task than <em>optimizeStrata</em>, but with a different Genetic Algorithm, operating on real values genome, instead of an integer one. This pemits to operate directly on the boundaries of the strata, instead of aggregating the initial atomic strata. In some situations (limited size of sampling frame) this new function is much more efficient. The limitation is in the nature of the stratification variables, that are required to be all continuous (though categorical ordinal could be handled).</p></li>
<li><p>A new function <em>expected_CV</em> has been added to calculate CVs on target variables in different domains that may be expected from a given solution, output of the <em>optimizeStrata</em> execution.</p></li>
<li><p>Fixed a bug in the optimization step when considering also take-all strata</p></li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.3" id="samplingstrata-13">SamplingStrata 1.3<small>2018-10-07</small><a class="anchor" aria-label="anchor" href="#samplingstrata-13"></a></h2>
<div class="section level3">
<h3 id="new-functions-1-3">New functions<a class="anchor" aria-label="anchor" href="#new-functions-1-3"></a></h3>
<ul><li><p>The optimization of population frame is run in parallel if different domains are considered. To this end, the parameter <em>parallel</em> can be set to TRUE in the <em>optimizeStrata</em> function. If not specified, n-1 of total available cores are used OR if number of domains &lt; (n-1) cores, then number of cores equal to number of domains are used.</p></li>
<li><p>A new function <em>KmeansSolution</em> produces an initial solution using the kmeans algorithm by clustering atomic strata considering the values of the means of target variables in them. Also, if the parameter <em>nstrata</em> is not indicated, the optimal number of clusters is determined inside each domain, and the overall solution is obtained by concatenating optimal clusters obtained in domains. By indicating this solution as a suggestion to the optimization step, this may greatly speed the convergence to the optimal solution.</p></li>
<li><p>A new function <em>selectSampleSystematic</em> has been added. It allows to select a stratified sample with the systematic method, that is a selection that begins selecting the first unit by an initial randomly chosen starting point, and proceeding in selecting other units by adding an interval that is the inverse of the sampling rate in the stratum. This selection method can be useful if associated to a particular ordering of the selection frame, where the ordering variable(s) can be considered as additional stratum variable(s).</p></li>
<li><p>It is now possible to handle <em>anticipated variance</em> by introducing a model linking a proxy variable whose values are available for all units in the sampling frame, with the target variable whose values are not available. In this implementation only linear model can be chosen. When calling the <em>buildStrataDF</em> function, a dataframe is given, containing three parameters (beta, sigma2 and gamma) for each couple target / proxy. On the basis of these parameters, means and standard deviations in sampling strata are calculated accordingly to given formulas.</p></li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.2" id="samplingstrata-12">SamplingStrata 1.2<small>2018-02-18</small><a class="anchor" aria-label="anchor" href="#samplingstrata-12"></a></h2>
<div class="section level3">
<h3 id="major-changes-1-2">Major changes<a class="anchor" aria-label="anchor" href="#major-changes-1-2"></a></h3>
<ul><li><p>The crossover function in the genetic algorithm has been modified by considering the <em>grouping</em> version of this algorithm: instead of mixing chromosomes in an indifferentiate way, groups of them in one parent (representing already aggregated strata) are attributed to the other parent when generating a child, preserving their composition. Moreover, parents are selected with a probability proportional to their fitness (in the previous version selection was completely at random). In many cases this can greatly speed the convergence to an optimal solution.</p></li>
<li><p>A new function <em>adjustSize</em> has been added. It allows to adjust the sample size and related allocation in strata on the basis of an externally indicated overall sample size. The adjustment of the sample size is perfomed by increasing or decreasing it proportionally in each optimized stratum.</p></li>
<li><p>A new function <em>buildFrameDF</em> has been added. It allows to create a <em>sampling frame</em> dataframe by indicating the dataset in which the information on all the units are contained, the identifier, the X variables, the Y variable and the variable that indicates the domains of interest.</p></li>
<li><p>In function <em>optimizeStrata</em> now the <em>initialStrata</em> parameter is a vector, whose length is equal to the number of strata in the different domains.</p></li>
<li><p>All outputs are written to an .subdirectory.</p></li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.1" id="samplingstrata-11">SamplingStrata 1.1<small>2016-01-13</small><a class="anchor" aria-label="anchor" href="#samplingstrata-11"></a></h2>
<div class="section level3">
<h3 id="major-changes-1-1">Major changes<a class="anchor" aria-label="anchor" href="#major-changes-1-1"></a></h3>
<ul><li><p>Function <em>memoise</em> from the same package (now required) is applied before each evaluation in order to save processing time. This may largely increase the efficiency of the algorithm.</p></li>
<li><p>A <em>recode</em> function is applied on every generated solution in order to recode a genotype of n genes with k&lt;=n distinct alleles 1, 2, …, k in such a way that the distinct alleles of the recoded genotype appear in the natural order 1, 2, …, k. This avoids to consider as distinct two solutions that are equivalent but make use of a different coding.</p></li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.0-4" id="samplingstrata-10-4">SamplingStrata 1.0-4<small>2015-08-25</small><a class="anchor" aria-label="anchor" href="#samplingstrata-10-4"></a></h2>
<div class="section level3">
<h3 id="bug-fix-for-old-releases-1-0-4">Bug fix for old releases<a class="anchor" aria-label="anchor" href="#bug-fix-for-old-releases-1-0-4"></a></h3>
</div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.0-3" id="samplingstrata-10-3">SamplingStrata 1.0-3<small>2014-10-21</small><a class="anchor" aria-label="anchor" href="#samplingstrata-10-3"></a></h2>
<div class="section level3">
<h3 id="new-functions-and-changes-1-0-3">New functions and changes<a class="anchor" aria-label="anchor" href="#new-functions-and-changes-1-0-3"></a></h3>
<ul><li><p>Modified the output to the console or to the file of results: of all the solutions, only the optimal value for each generation is visualised.</p></li>
<li><p>Now the visualisation of the trend in the optimal and mean values is optional: the plot can be avoided by setting showPlot = FALSE when calling optimizeStrata. It can be advisable when the number of iterations is very high.</p></li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.0-2" id="samplingstrata-10-2">SamplingStrata 1.0-2<small>2014-03-04</small><a class="anchor" aria-label="anchor" href="#samplingstrata-10-2"></a></h2>
<div class="section level3">
<h3 id="bug-fix-for-old-releases-1-0-2">Bug fix for old releases.<a class="anchor" aria-label="anchor" href="#bug-fix-for-old-releases-1-0-2"></a></h3>
</div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.0-1" id="samplingstrata-10-1">SamplingStrata 1.0-1<small>2013-01-16</small><a class="anchor" aria-label="anchor" href="#samplingstrata-10-1"></a></h2>
<div class="section level3">
<h3 id="major-changes-1-0-1">Major changes<a class="anchor" aria-label="anchor" href="#major-changes-1-0-1"></a></h3>
<ul><li><p>The object returned by function <em>optimizeStrata</em> is no more a dataframe but a list: (i) the first element of the list is the solution vector (solution<span class="math inline"><em>i</em><em>n</em><em>d</em><em>i</em><em>c</em><em>e</em><em>s</em>); (<em>i</em><em>i</em>)<em>t</em><em>h</em><em>e</em><em>s</em><em>e</em><em>c</em><em>o</em><em>n</em><em>d</em><em>e</em><em>l</em><em>e</em><em>m</em><em>e</em><em>n</em><em>t</em><em>o</em><em>f</em><em>t</em><em>h</em><em>e</em><em>l</em><em>i</em><em>s</em><em>t</em><em>i</em><em>s</em><em>t</em><em>h</em><em>e</em><em>d</em><em>a</em><em>t</em><em>a</em><em>f</em><em>r</em><em>a</em><em>m</em><em>e</em><em>c</em><em>o</em><em>n</em><em>t</em><em>a</em><em>i</em><em>n</em><em>i</em><em>n</em><em>g</em><em>a</em><em>g</em><em>g</em><em>r</em><em>e</em><em>g</em><em>a</em><em>t</em><em>e</em><em>d</em><em>s</em><em>t</em><em>r</em><em>a</em><em>t</em><em>a</em>(<em>s</em><em>o</em><em>l</em><em>u</em><em>t</em><em>i</em><em>o</em><em>n</em></span>aggr_strata).</p></li>
<li><p>In all the functions that previously produced .csv files and .pdf plots in the working directory, as a default this is no more the current behaviour. To write these files, it is now necessary to set the <em>writeFiles</em> flag to TRUE.</p></li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.0" id="samplingstrata-10">SamplingStrata 1.0<small>2012-12-04</small><a class="anchor" aria-label="anchor" href="#samplingstrata-10"></a></h2>
<div class="section level3">
<h3 id="major-changes-1-0">Major changes<a class="anchor" aria-label="anchor" href="#major-changes-1-0"></a></h3>
<ul><li>Two new functions: (i) <em>evalSolution</em>, to evaluate the found solution in terms of expected target variables precision and bias obtainable by samples drawn from the otpimized frame; (ii) <em>tuneParameters</em>, to determine the best combination of values to assign to the parameters necessary for the execution of the genetic algorithm used for the optimization of the frame stratification.</li>
</ul></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>

</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Giulio Barcaroli, Marco Ballin, Hanjo Odendaal, Daniela Pagliuca, Egon Willighagen, Diego Zardetto.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

